src/app/OperationalDeviceManager.h

```diff
@@ -71,35 +72,38 @@ class CASESessionManager : public Dnssd::ResolverDelegate
         return CHIP_NO_ERROR;
     }
 
-    virtual ~CASESessionManager() { mDNSResolver.Shutdown(); }
+    virtual ~OperationalDeviceManager() { mDNSResolver.Shutdown(); }
 
     /**
      * Find an existing session for the given node ID, or trigger a new session request.
      * The caller can optionally provide `onConnection` and `onFailure` callback objects. If provided,
      * these will be used to inform the caller about successful or failed connection establishment.
+     *
+     * The user shall call ReleaseDevice to free the device in the failure callback.
+     *
      * If the connection is already established, the `onConnection` callback will be immediately called.
      */
-    CHIP_ERROR FindOrEstablishSession(PeerId peerId, Callback::Callback<OnDeviceConnected> * onConnection,
-                                      Callback::Callback<OnDeviceConnectionFailure> * onFailure);
+    CHIP_ERROR AcquireDevice(PeerId peerId, Callback::Callback<OnDeviceConnected> * onConnection,
+                             Callback::Callback<OnDeviceConnectionFailure> * onFailure);
 
-    OperationalDeviceProxy * FindExistingSession(PeerId peerId);
+    void ReleaseDevice(OperationalDeviceProxy * device);
```

Comment:
If we're going to have a manager-like object, I'd very much prefer if we limited interactions with it to just during initialization/allocation of the `OperationalDeviceProxy`. There-after, clients should just be engaging with the latter, not the former.